module/VuFind/src/VuFind/Hierarchy/TreeRenderer/JSTree.php

```diff
@@ -249,7 +249,14 @@ protected function getUrlFromRouteCache($route, $id)
                     'recordID' => '__record_id__'
                 ]
             ];
-            $cache[$route] = $this->router->fromRoute($route, $params, $options);
+            $routeName = $route;
```

Comment:
I wonder if this new logic would actually fit better as a support method, both for readability and overriding... e.g.

<pre>
protected function getRouteNameFromDatasource($route)
{
    if ($route === 'collection') {
        return $this->getDataSource->getCollectionRoute();
    }
    if ($route === 'record') {
        return $this->getDataSource->getRecordRoute();
    }
    return $route;
}
</pre>

That's extremely nitpicky, and you can feel free to leave this as-is if you prefer... but that feels marginally better to me.

Reply:
I agree, more clear this way. It also provides even more flexibility for extending.